elrond-wasm-derive/src/parse/argument_parse.rs

```diff
@@ -76,3 +35,112 @@ fn missing_self_panic(m: &syn::TraitItemMethod) -> ! {
 		m.sig.ident.to_string()
 	)
 }
+
+fn extract_method_arg(pat_typed: &syn::PatType) -> MethodArgument {
+	let pat = &*pat_typed.pat;
+	let ty = &*pat_typed.ty;
+	let mut arg_metadata = ArgMetadata {
```

Comment:
Could maybe have a `new()` function for ArgMetadata that creates one with the default values you're using here.